src/app/pages/buyer/buyer-bid-list/buyer-bid-list.component.ts

```diff
@@ -1,23 +1,81 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { AppService } from 'app/app.service';
-import { Bid } from 'app/app.model';
+import { Bid, Buyer } from 'app/app.model';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'wm-buyer-bid-list',
   templateUrl: './buyer-bid-list.component.html',
   styleUrls: ['./buyer-bid-list.component.scss']
 })
-export class BuyerBidListComponent implements OnInit {
+export class BuyerBidListComponent implements OnInit, OnDestroy {
   public bids: Bid[] = [];
+  private routeParamSubscr: Subscription;
+  public buyer: Buyer;
 
-  constructor(public appServ: AppService) {}
+  constructor(public appServ: AppService, private route: ActivatedRoute, private router: Router) {}
 
   ngOnInit() {
-    this.appServ.getBidsForBuyer(this.appServ.loggedInUserInfo['id']).subscribe(data => {
+    this.listenToRouteParams();
+  }
+
+  /**
+   * @description listen to the route params for changes in buyer id
+   */
+  private listenToRouteParams() {
+    this.routeParamSubscr = this.route.paramMap.subscribe(params => {
+      let _buyerId: number = +params.get('id');
```

Comment:
@sriramj19 can we avoid these underscored variable names. i remember we liked it at a point. they are looking weird now.

Reply:
I have followed using underscored variable names for temporary and transient variables that are stored for reference of future usage. If removing that could make it understandable and readable, will remove it. Need your thoughts on this.

Reply:
Usually, underscores are used for private methods/variables. this is a local variable. better remove it.

Reply:
okay will do!